PetrolEngine.cpp

```diff
@@ -15,6 +16,47 @@ void PetrolEngine::changeGear(int gear)
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
+    
     std::cout << __FUNCTION__ << std::endl;
+    try
+    {
+        if (gear >= -1 && gear <= gears_) 
+        {
+            try
+            { 
+                if (currentGear_ == -1 && gear != 1)
+                {   
+//                  throw std::logic_error("Can't change gear from rear to higher than 1");
+                    throw 0;
+                }
+
+                if (currentGear_ > 1 && gear == -1) 
+                {
+                    throw 1;   
+//                  std::cout << "Can't change gear from " << currentGear_ << " to " << gear << std::endl;
+                }
+                
+                else {currentGear_ = gear;}
+
+            }
+            catch (...)
+            {
+                std::cout << "Logical exception" << std::endl;
+            }
+ 
+
+        }
+        else {throw std::out_of_range("Invalid gear!");}
+    }
+    catch (std::exception &e)
```

Comment:
lepiej tutaj nie łapać tych wyjątków. Jeśli już to w main. Poza tym jak rzucacie intami to tak ich nie złapiecie.